File: common/src/main/java/com/mthwate/conk/UpdateChecker.java
package com.mthwate.conk;

/**
 * @author mthwate
 */
public class UpdateChecker {

	private long lastUpdated;

	public UpdateChecker() {
		update();
	}

	public boolean hasUpdated(long lastCheck) {
		return lastCheck <= lastUpdated;
	}

	public void update() {
		lastUpdated = System.nanoTime();
	}

}

File: common/src/main/kotlin/com/mthwate/conk/UpdateChecker.kt
package com.mthwate.conk

/**
 * @author mthwate
 */
class UpdateChecker {

	private var lastUpdated: Long = 0

	init {
		update()
	}

	fun hasUpdated(lastCheck: Long): Boolean {
		return lastCheck <= lastUpdated
	}

	fun update() {
		lastUpdated = System.nanoTime()
	}

}